crates/cheatnet/src/cheatcodes/declare.rs

```diff
@@ -71,14 +69,8 @@ impl BlockifierState<'_> {
 }
 
 fn get_class_hash(sierra_contract: &str) -> Result<ClassHash> {
-    let sierra_class: ContractClass = serde_json::from_str(sierra_contract)?;
-    let abi_flattened = sierra_class.abi.unwrap().json();
-    let mut sierra_contract_map: HashMap<String, Value> = serde_json::from_str(sierra_contract)?;
-    sierra_contract_map.insert("abi".to_string(), Value::String(abi_flattened));
-    let sierra_contract = serde_json::to_string_pretty(&sierra_contract_map)?;
-
-    let sierra_class: FlattenedSierraClass = serde_json::from_str(&sierra_contract)?;
-    let class_hash = sierra_class.class_hash();
```

Comment:
What happened here before? Did we extract abi and then insert it again? Does anyone have an idea?

Reply:
cc @cptartur

Reply:
Scarb artifacts represent ABIs as json structures, where blockifier expected the ABI to be a flat string (this is how actual network works as well for cairo1 contracts), so I had to flatten the abi, otherwise serialization would fail.

I see a different type is used for sierra class now that seems to accept abis like this.